src/Toml/Parser.hs

```diff
@@ -89,10 +89,11 @@ quote q t = q <> t <> q
 nonControlChar :: Parser Text
 nonControlChar = Text.singleton <$> satisfy (not . isControl)
 
-multilineStringP :: Parser Text
-multilineStringP = lexeme $ fmap mconcat $ quotes3
-                                        >> optional eol
-                                        >> allowedChar `manyTill` quotes3
+multilineP :: Parser Text -> Parser Text -> Parser [Text]
+multilineP quotes allowedChar = quotes >> optional eol >> allowedChar `manyTill` quotes
+
+multilineBasicStringP :: Parser Text
+multilineBasicStringP = lexeme $ fmap mconcat $ multilineP quotes3 allowedChar
```

Comment:
You can move `lexeme $ fmap mconcat $` into `multilineP` as well to remove even more code duplication.